admin/templates/project/streamlit.yaml

```diff
@@ -0,0 +1,82 @@
+#!jinja2: lstrip_blocks: True
+{#
+    REQUIRED APPLICATION VARIABLES
+    ------------------------------
+    image : Docker image path
+    port  : Container port which public HTTP service runs on
+    replicas: how many instances of the application to run
+#}
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: streamlit-app
+  namespace: {{ inventory_hostname }}
+  labels:
+    app: {{ inventory_hostname }}
+spec:
+  replicas: {{ app.replicas }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: {{ inventory_hostname }}
+  template:
+    metadata:
+      labels:
+        app: {{ inventory_hostname }}
+    spec:
+      containers:
+      - image: {{ app.image }}
+        name: {{ inventory_hostname }}
+        ports:
+        - containerPort: {{ app.port }}
+          name: http
+          protocol: TCP
+      imagePullSecrets:
+      - name: regcred
```

Comment:
This is pure copy/paste, no clue if it is appropriate here

Reply:
This line enables us to deploy from our private in-cluster docker registry, useful if we need to push image builds that we don't want to post to dockerhub. I think in this case we'd be fine publishing the container image we deploy from via dockerhub as it doesn't seem we need to tailor it at all to our deploy